File: control-assurance-sp/src/types/GIAAImport.ts
import { IEntity } from "./Entity";

export interface IGIAAImport  extends IEntity {

    XMLContents?: string;
    Status?: string;
    LogDetails?: string;
    LastImportStatus?: string;
    LastImportDate?: Date;
    LastImportById?: number;
}

export interface IGIAAImportInfo extends IEntity {
    Status?: string;
    LogDetails?: string;
    LastImportStatus?: string;
    LastImportDate?: string;
    LastImportBy?: string;
}

export class GIAAImport implements IGIAAImport{ 
    public ID: number = 0;
    public Title: string = null;
    public XMLContents?: string = null;
    public Status?: string = null;
    public LogDetails?: string = null;
    public LastImportStatus?: string = null;
    public LastImportDate?: Date = null;
    public LastImportById?: number = null;
}

File: ControlAssuranceAPI/Models/GIAAImport.cs
using System;
using System.Collections.Generic;

namespace CAT.Models;

public partial class GIAAImport
{
    public int ID { get; set; }

    public string? Title { get; set; }

    public string? XMLContents { get; set; }

    public string? Status { get; set; }

    public string? LogDetails { get; set; }

    public string? LastImportStatus { get; set; }

    public DateTime? LastImportDate { get; set; }

    public int? LastImportById { get; set; }

    public virtual User? LastImportBy { get; set; }
}
public class GIAAImportInfoView_Result
{
    public int ID { get; set; }
    public string Status { get; set; } = "";
    public string LogDetails { get; set; } = "";
    public string LastImportStatus { get; set; } = "";
    public string LastImportDate { get; set; } = "";
    public string LastImportBy { get; set; } = "";
}
